### PR TITLE
Add `minion_manager.rpc.client.py/tasks.py` unit tests

### PR DESCRIPTION
**Pull request by mihaelabalutoiu**
This PR adds test cases for the following module:

- `minion_manager.rpc.tasks.py`
- `minion_manager.rpc.client.py`

### CODE REVIEW

**Review by Dany9966 — CHANGES_REQUESTED** [review]
First review pass done! Please look out for messages (log, event, progress updates, whatever) we do not want to include in tests, and remove all of them please.

**Review by mihaelabalutoiu — COMMENTED** [review]
I updated the code, please take a look now. Thank you!

File: coriolis/tests/minion_manager/rpc/test_tasks.py [hunk 1]
```diff
@@ -0,0 +1,2009 @@
...
+        test_base.CoriolisBaseTestCase):
+
+    def setUp(self):
+        super(ConfirmMinionAllocationForMigrationTaskTestCase, self).setUp()
+        self.action_id = mock.sentinel.action_id
+        self.allocated_machine_id_mappings = [
+            mock.sentinel.allocate_machine_id_mappings]
+
+        self.task = tasks.ConfirmMinionAllocationForMigrationTask(
+            mock.sentinel.action_id, self.allocated_machine_id_mappings)
```
Comment (Dany9966) [inline coriolis/tests/minion_manager/rpc/test_tasks.py:487]: You don't use `self.allocated_machine_id_mappings` anywhere. You can pass that sentinel directly
```suggestion
        self.task = tasks.ConfirmMinionAllocationForMigrationTask(
            mock.sentinel.action_id,
            mock.sentinel.allocated_machine_id_mappings)
```

File: coriolis/tests/minion_manager/rpc/test_tasks.py [hunk 2]
```diff
@@ -0,0 +1,2009 @@
...
+class ConfirmMinionAllocationForReplicaTaskTestCase(
+        test_base.CoriolisBaseTestCase):
+
+    def setUp(self):
+        super(ConfirmMinionAllocationForReplicaTaskTestCase, self).setUp()
+        self.allocated_machine_id_mappings = [
+            mock.sentinel.allocate_machine_id_mappings]
+
+        self.task = tasks.ConfirmMinionAllocationForReplicaTask(
+            mock.sentinel.action_id, self.allocated_machine_id_mappings)
```
Comment (Dany9966) [inline coriolis/tests/minion_manager/rpc/test_tasks.py:526]: Same as above

**Review by Dany9966 — CHANGES_REQUESTED** [review]

**Comment by Dany9966** [comment]
Mostly lgtm, you only have a little missed case in `HealthcheckMinionMachineTask.execute`. Please handle that too.

File: coriolis/tests/minion_manager/rpc/test_tasks.py [hunk 3]
```diff
@@ -0,0 +1,2038 @@
...
+    @mock.patch.object(tasks.BaseMinionManangerTask, 'execute')
+    @mock.patch.object(
+        tasks.MinionManagerTaskEventMixin,
+        '_set_minion_machine_allocation_status'
+    )
+    def test_execute_with_exception_and_fail_on_error(
+            self, mock_set_minion_machine_allocation, mock_execute,
+            mock_add_minion_pool_event, mock_get_minion_machine):
+        mock_get_minion_machine.return_value = self.minion_machine
+        mock_execute.side_effect = CoriolisTestException
```
Comment (Dany9966) [inline coriolis/tests/minion_manager/rpc/test_tasks.py:1679]: I don't get what you're testing here. You mock the tested method, make it raise and then check that the raise has occurred? 
For this case, you need to:
1. set `mock_get_minion_machine.return_value = None`;
2. remove the `mock_execute` entirely;
3. keep `_fail_on_error` on `True`;
4. Check that the call raises `InvalidMinionMachineState`.

**Review by Dany9966 — CHANGES_REQUESTED** [review]

File: coriolis/tests/minion_manager/rpc/test_tasks.py (hunk 3 above)
Comment (Dany9966) [reply coriolis/tests/minion_manager/rpc/test_tasks.py:1679]: Nevermind this, this case was already covered. This case is also good. Thank you!

**Review by Dany9966 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**